### PR TITLE
Use exceptions and a little more OO

### PR DESCRIPTION
**Pull request by weshooper**
Throwing exceptions makes it easier to work within controllers, but makes it a breaking change so this'll need to be a version 2.0.0.

I've also:

- Split out passing recent hashes into own function
- Added a convenience function for checking a password confirmation if that's something we need
- Add a README, with examples
- Added a few more test situations

### CODE REVIEW

**Comment by ziadoz** [comment]
👍

File: src/PasswordChecker.php [hunk 1]
```diff
@@ -50,19 +63,14 @@ private function isPasswordObvious(string $password): bool
         return is_numeric(str_replace([' ', '/', '-'], '', $password)); // Could be a DOB or phone number
     }
 
-    private function isRecentPassword(string $password, array $recentHashes = []): bool
+    private function isRecentPassword(string $password): bool
     {
-        foreach ($recentHashes as $hash) {
+        foreach ($this->recentHashes as $hash) {
```
Comment (maxakropolis) [inline src/PasswordChecker.php:68]: Will this fail if `setPreviousPasswords` isn't used?

Comment (weshooper) [reply src/PasswordChecker.php:68]: @maxakropolis good point ta. Verified, added a test and fix...

File: README.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+```
+$checker = new PasswordChecker;
+$checker->validate('abc123'); // throws PasswordException (too short)
+$checker->validate('password123'); // throws PasswordException (too common)
+$checker->validate('123-456-7890'); // throws PasswordException (too numeric)
+$checker->validate('31/12/1999'); // throws PasswordException (too numeric)
+$checker->validate('we love php'); // returns true
+```
+
+That's it. Though you can add further (optional, but recommended) checks and restrictions...
```
Comment (WillJW) [inline README.md]: I don't think an ellipsis is correct here.

Comment (weshooper) [reply README.md]: @WillJW Cheers, have removed.

File: tests/PasswordCheckerTest.php [hunk 3]
```diff
@@ -4,79 +4,131 @@
...
+    /**
+     * @expectedException        FusionsPim\PhpPasswordChecker\PasswordException
+     * @expectedExceptionMessage New password must be at least 10 characters long
+     */
     public function test_fails_due_to_short_password()
     {
-        $checker = new PasswordChecker(static::APP_SPECIFIC_OBVIOUS_PASSWORDS);
-        $this->assertFalse($checker->validate('abc', static::USER_PASSWORD_HISTORY));
-        $this->assertEquals('New password must be at least 10 characters long', $checker->getErrorMessage());
+        $this->checker->validate('abc');
```
Comment (WillJW) [inline tests/PasswordCheckerTest.php:42]: Is it worth adding a test using a multibyte string that would be >10 characters when using standard `strlen`?

**Review by WillJW — APPROVED** [review]
Other than minor comments, looking good! 👍

File: tests/PasswordCheckerTest.php (hunk 3 above)
Comment (weshooper) [reply tests/PasswordCheckerTest.php:42]: @WillJW done ta

**Comment by maxakropolis** [comment]
👍

**Pull request merged**